### ISSUE TITLE
estimateD output malformed at certain target coverage levels

### ISSUE THREAD

**Issue by rclose** [JohnsonHsieh/iNEXT#12]
I just tried to use estimateD on the following abundance vector:

```
estimateD(x = c(34, 18, 14, 2, 1, 1, 1, 1), datatype = "abundance", level = 0.4, base = "coverage")
```

At certain coverage levels, such as 0.4, the output is malformed, containing twice as many rows as necessary and spurious diversity estimates: 

```
  m       method order    SC qD qD.LCL qD.UCL
1 1 interpolated     0 0.315  1      1      1
2 1 interpolated     0 0.315  1      1      1
3 1 interpolated     1 0.315  1      1      1
4 1 interpolated     1 0.315  1      1      1
5 1 interpolated     2 0.315  1      1      1
6 1 interpolated     2 0.315  1      1      1

```

Whereas at higher or very low coverage levels, such as 0.6 or 0.1, the output looks correct:

```
  m       method order    SC    qD qD.LCL qD.UCL
1 3 interpolated     0 0.635 2.174  2.021  2.327
2 3 interpolated     1 0.635 2.006  1.847  2.164
3 3 interpolated     2 0.635 1.840  1.674  2.006

```

Any idea what's going on?

**Comment by JohnsonHsieh** [comment]
The compute method for estimateD() is choose the nearest sample size (an integer) for the given level. In your case (plz see the following), the nearest sample size is m = 1. 

m, sc
1, 0.3153365
2, 0.5105522
3, 0.6346728

Do you think that whether researcher and iNEXT user need a estimateD_plus function to compute float number for given level ? (such as 1.29347 for level = 0.4)

BR
Johnson

**Comment by rclose** [comment]
Hi Johnson,

Thanks for identifying the reason behind this issue. Personally, I think an estimateD_plus function that gave diversity with decimal places would be very useful. 

That said, I would expect estimateD to present the results for a sample coverage of 0.4 like this:
```
 m       method order    SC qD qD.LCL qD.UCL
1 1 interpolated     0 0.315  1      1      1
2 1 interpolated     1 0.315  1      1      1
3 1 interpolated     2 0.315  1      1      1
```

Am I right to think that the current behaviour of outputting duplicate rows for each q number is a bug?

**Comment by JohnsonHsieh** [comment]
Hi rclose,

It's not a bug.

For m = 1, it means there is a rarefied assemblage contains only one individual. Thus this assemblage  contains only on species with relative abundance p = 1. In this case, Hill numbers obviously equal to 1 for all order q.

**Comment by rclose** [comment]
Hi Johnson,

Yes, I agree that diversity should be 1 for all orders q. But if you look at my first post in this thread, you'll see that each order q is reported *twice*—there are six rows in the matrix when there should only be three. This is what I meant by 'malformed' output. 

Many thanks,
Roger

**Comment by JohnsonHsieh** [comment]
got it, thanks Roger. It has been fixed in iNEXT version 2.0.12.

```
 estimateD(x = c(34, 18, 14, 2, 1, 1, 1, 1), datatype = "abundance", level = 1, base = "size") 
```
```
  m       method order    SC qD qD.LCL qD.UCL
1 1 interpolated     0 0.315  1      1      1
2 1 interpolated     1 0.315  1      1      1
3 1 interpolated     2 0.315  1      1      1
```
